### PR TITLE
wip: tx spec

### PR DESCRIPTION
**Pull request by jxom**

### CODE REVIEW

**Comment by gakonst** [comment]
ref https://www.notion.so/warpcast/Frames-Transactions-Public-Draft-c2e0d3210d684b4cb7803de1810db36d?pvs=4#96d3bcdae85040e69967de9696cd3eaf

File: playground/src/index.tsx [hunk 1]
```diff
@@ -207,5 +207,234 @@ app.frame('/image-only', (c) => {
...
+    to: '0x00000000fcCe7f938e7aE6D3c335bD6a1a7c593D',
+    value: '984316556204476',
+    chainId: '10',
+  })
+})
+
 app.route('/todos', todoApp)
 app.route('/routing', routingApp)
+
+export const erc20Abi = [
```
Comment (gakonst) [inline playground/src/index.tsx]: we should viemify these, human readable abi etc.

File: src/components/Button.tsx [hunk 2]
```diff
@@ -120,9 +120,32 @@ export function ButtonReset({
...
+}: experimental_ButtonTransactionProps) {
+  return [
+    <meta
+      property={`fc:frame:button:${index}`}
+      content={children}
+      data-value="_t"
+    />,
+    <meta property={`fc:frame:button:${index}:action`} content="tx" />,
+    <meta property={`fc:frame:button:${index}:target`} content={location} />,
+  ] as unknown as HtmlEscapedString
```
Comment (gakonst) [inline src/components/Button.tsx]: clean

File: src/types/utils.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  ///
+  keys extends KeyofUnion<union> = KeyofUnion<union>,
+> = union extends infer Item
+  ? Pretty<Item & { [_K in Exclude<keys, keyof Item>]?: undefined }>
+  : never
+type KeyofUnion<type> = type extends type ? keyof type : never
+
+export type Pretty<type> = { [key in keyof type]: type[key] } & unknown
+
+export type Widen<type> =
```
Comment (gakonst) [inline src/types/utils.ts:14]: magic

File: playground/src/index.tsx [hunk 4]
```diff
@@ -207,5 +207,234 @@ app.frame('/image-only', (c) => {
...
+  }
+})
+
+app.experimental_transaction('/tx', (c) => {
+  return c.res({
+    description: 'Rent 1 Farcaster storage unit to FID 3621',
+    to: '0x00000000fcCe7f938e7aE6D3c335bD6a1a7c593D',
+    data: '0x783a112b0000000000000000000000000000000000000000000000000000000000000e250000000000000000000000000000000000000000000000000000000000000001',
+    value: '984316556204476',
+    chainId: '10',
```
Comment (dalechyn) [inline playground/src/index.tsx]: Shouldn't `value` be a `bigint` and `chainId` a `number`?

**Review by dalechyn — CHANGES_REQUESTED** [review]
Typings question.

File: src/components/Button.tsx [hunk 5]
```diff
@@ -119,11 +119,34 @@ export function ButtonReset({
   )
 }
 
+export type ButtonTransactionProps = ButtonProps & {
+  location: string
```
Comment (tmm) [inline src/components/Button.tsx:123]: Does it make sense to use `target` instead to keep inline with the spec?

**Comment by jxom** [comment]
Succeeded by #63.

**Pull request closed without merging**